pkg/intermediate/aggregate.go

```diff
@@ -144,92 +144,83 @@ func (a *aggregation) removeDuplicates(tuple Tuple) {
-		srcIPAddr, ok := infoElementMap["sourceIPv4Address"].Value.(net.IP)
-		if !ok {
-			return Tuple{}, fmt.Errorf("sourceIPv4Address is not in correct format.")
-		}
-		srcIP = srcIPAddr.String()
-		dstIPAddr, ok := infoElementMap["destinationIPv4Address"].Value.(net.IP)
-		if !ok {
-			return Tuple{}, fmt.Errorf("destinationIPv4Address is not in correct format.")
-		}
-		dstIP = dstIPAddr.String()
-		srcPortNum, ok := infoElementMap["sourceTransportPort"].Value.(uint16)
-		if !ok {
-			return Tuple{}, fmt.Errorf("sourceTransportPort is not in correct format.")
-		}
-		srcPort = srcPortNum
-		dstPortNum, ok := infoElementMap["destinationTransportPort"].Value.(uint16)
-		if !ok {
-			return Tuple{}, fmt.Errorf("destinationTransportPort is not in correct format.")
-		}
-		dstPort = dstPortNum
-		protoNum, ok := infoElementMap["protocolIdentifier"].Value.(uint8)
-		if !ok {
-			return Tuple{}, fmt.Errorf("protocolIdentifier is not in correct format.")
-		}
-		proto = protoNum
-	} else if containsInfoElement(infoElementMap, "sourceIPv6Address") && containsInfoElement(infoElementMap, "destinationIPv6Address") && containsInfoElement(infoElementMap, "sourceTransportPort") && containsInfoElement(infoElementMap, "destinationTransportPort") && containsInfoElement(infoElementMap, "protocolIdentifier") {
-		srcIPAddr, ok := infoElementMap["sourceIPv6Address"].Value.(net.IP)
-		if !ok {
-			return Tuple{}, fmt.Errorf("sourceIPv6Address is not in correct format.")
-		}
-		srcIP = net.IP(srcIPAddr).String()
-		dstIPAddr, ok := infoElementMap["destinationIPv6Address"].Value.(net.IP)
-		if !ok {
-			return Tuple{}, fmt.Errorf("destinationIPv6Address is not in correct format.")
-		}
-		dstIP = net.IP(dstIPAddr).String()
-		srcPortNum, ok := infoElementMap["sourceTransportPort"].Value.(uint16)
-		if !ok {
-			return Tuple{}, fmt.Errorf("sourceTransportPort is not in correct format.")
-		}
-		srcPort = srcPortNum
-		dstPortNum, ok := infoElementMap["destinationTransportPort"].Value.(uint16)
-		if !ok {
-			return Tuple{}, fmt.Errorf("destinationTransportPort is not in correct format.")
+// getFlowKeyFromRecord returns 5-tuple from data record
+func getFlowKeyFromRecord(record entities.Record) (FlowKey, error) {
+	srcPortElement, exist := record.GetInfoElementWithValue("sourceTransportPort")
+	if !exist {
+		return FlowKey{}, fmt.Errorf("sourceTransportPort does not exist")
+	}
+	srcPort, ok := srcPortElement.Value.(uint16)
+	if !ok {
+		return FlowKey{}, fmt.Errorf("sourceTransportPort is not in correct format")
+	}
+
+	dstPortElement, exist := record.GetInfoElementWithValue("destinationTransportPort")
+	if !exist {
+		return FlowKey{}, fmt.Errorf("destinationTransportPort does not exist")
+	}
+	dstPort, ok := dstPortElement.Value.(uint16)
+	if !ok {
+		return FlowKey{}, fmt.Errorf("destinationTransportPort is not in correct format")
+	}
+
```

Comment:
Can these lines be simplified for code (including till L221) readability purposes? 
Define the flow key elements in string list, and then iterate over them using switch-case?
There is common code for (srcPort, dstPort) and (srcIP, dstIP). This decreases the lines of code.
What do you think?

Reply:
Agree. Changed to switch-case. Thanks!